### PR TITLE
[Enhancement] Reduce calling counts when refreshing Iceberg table

### PR DESCRIPTION
**Pull request by mxdzs0612**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

See #10756.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 15 / 31 (48.39%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|0|2|00.00%|[438, 439]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergHiveCatalog.java|0|2|00.00%|[84, 85]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|5|14|35.71%|[136, 146, 147, 149, 150, 151, 157, 158, 159]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergRepository.java|6|9|66.67%|[35, 36, 37]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 16 / 31 (51.61%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|0|2|00.00%|[438, 439]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergHiveCatalog.java|0|2|00.00%|[84, 85]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|5|13|38.46%|[135, 145, 146, 148, 149, 150, 156, 157]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergRepository.java|6|9|66.67%|[35, 36, 37]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|2|2|100.00%|[]|

**Comment by mxdzs0612** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 1]
```diff
@@ -1552,6 +1552,18 @@ public class Config extends ConfigBase {
     @ConfField(mutable = true)
     public static long iceberg_worker_num_threads = 64;
 
+    /**
+     * size of iceberg table refresh pool
+     */
+    @ConfField(mutable = true)
+    public static int iceberg_table_refresh_threads = 128;
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:1559]: why use 128?

Comment (mxdzs0612) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java:1559]: @caneGuy

Comment (mxdzs0612) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java:1559]: @caneGuy Any concerns or suggestions?

**Review by caneGuy — DISMISSED** [review]

**Review by DorianZheng — DISMISSED** [review]

**Review by stephen-shelby — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 16 / 31 (51.61%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|0|2|00.00%|[438, 439]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergHiveCatalog.java|0|2|00.00%|[84, 85]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|5|13|38.46%|[133, 143, 144, 146, 147, 148, 154, 155]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergRepository.java|6|9|66.67%|[35, 36, 37]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|2|2|100.00%|[]|

**Comment by mxdzs0612** [comment]
run starrocks_admit_test

**Comment by mxdzs0612** [comment]
run starrocks_admit_test

**Comment by mxdzs0612** [comment]
run starrocks_admit_test

**Review by caneGuy — DISMISSED** [review]

**Review by stephen-shelby — DISMISSED** [review]

**Review by dirtysalt — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 16 / 31 (51.61%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|0|2|00.00%|[439, 440]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergHiveCatalog.java|0|2|00.00%|[84, 85]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|5|13|38.46%|[133, 143, 144, 146, 147, 148, 154, 155]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergRepository.java|6|9|66.67%|[35, 36, 37]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|2|2|100.00%|[]|

**Review by caneGuy — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**